underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/EntityGroupFilterBuilder.java

```diff
@@ -44,25 +43,27 @@ protected List<String> entityGroupIds() {
   }
 
   @Override
-  protected Map<Literal, String> selectedIdsAndEntityGroups(String serializedSelectionData) {
-    Map<Literal, String> idsAndEntityGroups = new HashMap<>();
+  protected List<SelectionItem> selectedIdsAndGroups(String serializedSelectionData) {
+    List<SelectionItem> idsAndGroups = new ArrayList<>();
     DTEntityGroup.EntityGroup selectionData = deserializeData(serializedSelectionData);
     if (selectionData != null) {
       for (DTEntityGroup.EntityGroup.Selection selectedId : selectionData.getSelectedList()) {
         if (selectedId.hasKey()) {
-          idsAndEntityGroups.put(
-              Literal.forInt64(selectedId.getKey().getInt64Key()), selectedId.getEntityGroup());
+          ValueDataOuterClass.ValueData valueData =
+              selectedId.hasValueData() ? selectedId.getValueData() : null;
```

Comment:
curious: what is the expected behavior/ return value if a get is performed and it has no value